backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/NonRootManagedBranchSnapshot.java

```diff
@@ -70,14 +69,14 @@ void setParent(IManagedBranchSnapshot givenParentBranch) {
   }
 
   @Override
-  public Option<IForkPointCommitOfManagedBranch> getForkPoint() {
-    return Option.of(forkPoint);
+  public @Nullable IForkPointCommitOfManagedBranch getForkPoint() {
+    return forkPoint;
   }
 
   @Override
   public IGitRebaseParameters getParametersForRebaseOntoParent() throws GitMacheteMissingForkPointException {
     LOG.debug(() -> "Entering: branch = '${getName()}'");
-    if (forkPoint == null) {
+    if (forkPoint == null) { // in this place we are producing exception for null
```

Comment:
That might be better resolved with sth like:

https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/RequiresNonNull.html

`@RequiresNonNull({"forkPoint"})`

(and this change can be done regardless of whether we ditch `Options`)